agent/src/com/jonnyzzz/teamcity/virtual/VMDetector.java

```diff
@@ -46,44 +47,68 @@ public void beforeAgentConfigurationLoaded(@NotNull BuildAgent agent) {
 
       private void detectVagrant(@NotNull final BuildAgentConfiguration config) {
 
-        final String output = executeCommandWithShell("vagrant", "vagrant --version");
+        final String[] output = executeCommandWithShell("vagrant", "vagrant --version");
         if (output == null) return;
 
-        String ver = output.toLowerCase().trim();
+        String ver = output[0].toLowerCase().trim();
         if (ver.startsWith("vagrant")) ver = ver.substring("vagrant".length()).trim();
 
         if (StringUtil.isEmptyOrSpaces(ver)) {
-          LOG.warn("Failed to parse vagrant version: " + output);
+          LOG.warn("Failed to parse vagrant version: " + Arrays.toString(output));
           return;
         }
 
         config.addConfigurationParameter(VMConstants.VAGRANT_PROPERTY, ver);
       }
 
       private void detectDocker(@NotNull final BuildAgentConfiguration config) {
-        if (!SystemInfo.isLinux) {
-          LOG.debug("Docker is only available under Linux");
-          return;
-        }
 
-        final String output = executeCommandWithShell("docker", "docker --version");
+        final String[] output = executeCommandWithShell("docker", "docker version");
         if (output == null) return;
 
-        String ver = output.toLowerCase().trim();
-        ver = ver.replaceAll("\\s*docker\\s+version\\s+", "");
-        ver = ver.replaceAll(",?\\s+build\\s+", "-");
+        String[] cleanedArray = new String[output.length];
```

Comment:
Not relevant after this [commit](https://github.com/jonnyzzz/TeamCity.Virtual/pull/51/commits/7da68b403fe776c8f829420c75889f9981a73f65)